test/unit/host_test.rb

```diff
@@ -2137,6 +2137,13 @@ def to_managed!
     assert_equal 'foo', host.shortname
   end
 
+  test 'check operatingsystem and architecture association' do
+    primary = FactoryGirl.build(:nic_managed, :primary => true, :name => 'foo')
+    host = FactoryGirl.build(:host, :interfaces => [primary, FactoryGirl.build(:nic_managed, :provision => true)])
+    host.populate_fields_from_facts(:architecture => "x86_64", :operatingsystem => 'RedHat', :operatingsystemrelease => '6.2')
+    assert host.operatingsystem.architectures.include?(host.architecture), "no association between operatingsystem and architecture"
```

Comment:
Fixtures could be dangerous in this test, as the fact importers will find existing models.  I think you've avoided it by using "6.2" as the OS version, so I'd suggest wrapping the `populate_fields_from_facts` call in an `assert_difference` to check a new Operatingsystem was created, so it doesn't accidentally break later.